lighthouse-core/audits/dobetterweb/has-inspector-issues.js

```diff
@@ -0,0 +1,246 @@
+    'main thread for more than 60 seconds in total',
+  /** Message shown in a data table when the item is a HeavyAds issue where an ad has used the main thread for more than 15 seconds in any 30 second window. */
+  heavyAdsCPUPeakLimitMessage: 'The page contains ads that use the ' +
+    'main thread for more than 15 seconds in a 30 second window',
+  /** Message shown in a data table when the item is a ContentSecurityPolicy issue where resources are blocked due to not being in the Content Security Policy header. */
+  cspUrlViolationMessage: 'The `Content Security Policy` of the page blocks some resources because ' +
+    'their origin is not included in the content security policy header',
+  /** Message shown in a data table when the item is a ContentSecurityPolicy issue where the Content Security Policy blocks inline execution of scripts and stylesheets. */
+  cspInlineViolationMessage: 'The `Content Security Policy` of the page blocks ' +
+    'inline execution of scripts and stylesheets',
+  /** Message shown in a data table when the item is a ContentSecurityPolicy issue where the Content Security Policy blocks the use of the `eval` function in Javascript. */
+  cspEvalViolationMessage: 'The `Content Security Policy` of the site blocks ' +
+    'the use of `eval` in JavaScript',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+/** @type {Record<string, string>} */
+const heavyAdsMsgMap = {
+  'NetworkTotalLimit': str_(UIStrings.heavyAdsNetworkLimitMessage),
+  'CpuTotalLimit': str_(UIStrings.heavyAdsCPUTotalLimitMessage),
+  'CpuPeakLimit': str_(UIStrings.heavyAdsCPUPeakLimitMessage),
+};
+/** @type {Object<string, string>} */
+const contentSecurityPolicyMsgMap = {
+  'kInlineViolation': str_(UIStrings.cspInlineViolationMessage),
+  'kEvalViolation': str_(UIStrings.cspEvalViolationMessage),
+  'kURLViolation': str_(UIStrings.cspUrlViolationMessage),
+};
+/** @type {Object<string, string>} */
+const blockedByResponseMsgMap = {
+  'CoepFrameResourceNeedsCoepHeader': str_(UIStrings.coepResourceBlockedMessage),
+  'CoopSandboxedIFrameCannotNavigateToCoopPage': str_(UIStrings.coopIframeBlockedMessage),
+  'CorpNotSameOrigin': str_(UIStrings.coepResourceBlockedMessage),
+  'CorpNotSameOriginAfterDefaultedToSameOriginByCoep': str_(UIStrings.coepFrameBlockedMessage),
+  'CorpNotSameSite': str_(UIStrings.coepResourceBlockedMessage),
+};
+
+class IssuesPanelEntries extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'has-inspector-issues',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['InspectorIssues'],
+    };
+  }
+
+  /**
+   * @param {Array<LH.Crdp.Audits.MixedContentIssueDetails>} mixedContentIssues
+   * @return {Array<IssueItem>}
+   */
+  static getMixedContentItems(mixedContentIssues) {
+    if (!mixedContentIssues) {
```

Comment:
The artifact and the `param` declarations for all of these always have the arrays defined (defaulting to the empty array), so no need for these fallbacks